### PR TITLE
Issue 71614: Protect log message maps

### LINKED ISSUE #71614: kubelet panic: concurrent map write
**What happened**: kubelet paniced due to a concurrent map write

**What you expected to happen**: kubelet should not panic

**How to reproduce it (as minimally and precisely as possible)**: I'm seeing this running hack/ci/e2e.sh in https://sigs.k8s.io/kind

**Anything else we need to know?**: Logs are here: 

https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-kind-conformance-parallel/4622

https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-kind-conformance-parallel/4622/artifacts/logs/kind-1-control-plane/kubelet.log

**Environment**: `kind` cluster (kubernetes running in a docker container)
- Kubernetes version (use `kubectl version`):  `v1.14.0-alpha.0.752+3156df3e806cb8`
- Cloud provider or hardware configuration: n/a
- OS (e.g. from /etc/os-release): Ubuntu (in a container)
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig node

/kind bug

### PR DESCRIPTION
**Pull request by RobertKrawitz**
/kind bug

Fixes #71614 

**Special notes for your reviewer**: Check that locking logic is correct

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/remote/remote_runtime.go [hunk 1]
```diff
@@ -40,6 +41,7 @@ type RemoteRuntimeService struct {
 	lastError map[string]string
 	// Time last per-container error message was printed
 	errorPrinted map[string]time.Time
+	errorMapLock sync.RWMutex
```
[the remark this answers is not in the document]
Comment (RobertKrawitz) [reply pkg/kubelet/remote/remote_runtime.go]: Yep.

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 2]
```diff
@@ -131,6 +132,7 @@ type kubeGenericRuntimeManager struct {
 
 	// Time last per-container error message was printed
 	errorPrinted map[string]time.Time
+	errorMapLock sync.RWMutex
```
[the remark this answers is not in the document]
Comment (RobertKrawitz) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: Yep.

**Comment by BenTheElder** [comment]
/priority critical-urgent
feel free to change, but imo this is a breaking bug due to the panic, and it needs to have a priority set for merge so

**Comment by sjenning** [comment]
/lgtm
Looking back at this now, would have been nice to abstract all this so it isn't so much code inline on these paths :-/

**Comment by yujuhong** [comment]
> Looking back at this now, would have been nice to abstract all this so it isn't so much code inline on these paths

+1 This code is brittle. Merging this to unblock the tests.

/approve

@RobertKrawitz could you follow up with a separate PR to clean this up? Thanks

**Comment by RobertKrawitz** [comment]
@yujuhong will do

**Comment by BenTheElder** [comment]
Thanks all!

**Comment by RobertKrawitz** [comment]
@yujuhong See https://github.com/kubernetes/kubernetes/pull/71679

**Pull request merged**